grpc-circuitbreaker-utils/src/main/java/org/hypertrace/circuitbreaker/grpcutils/resilience/ResilienceCircuitBreakerInterceptor.java

```diff
@@ -0,0 +1,137 @@
+package org.hypertrace.circuitbreaker.grpcutils.resilience;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Singleton;
+import com.typesafe.config.Config;
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerConfig;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ForwardingClientCallListener;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.circuitbreaker.grpcutils.CircuitBreakerConfigProvider;
+import org.hypertrace.circuitbreaker.grpcutils.CircuitBreakerInterceptor;
+
+@Slf4j
+@Singleton
+public class ResilienceCircuitBreakerInterceptor extends CircuitBreakerInterceptor {
+
+  public static final CallOptions.Key<String> CIRCUIT_BREAKER_KEY =
+      CallOptions.Key.createWithDefault("circuitBreakerKey", "default");
+  private final CircuitBreakerRegistry resilicenceCircuitBreakerRegistry;
+  private final CircuitBreakerConfigProvider circuitBreakerConfigProvider;
+  private final Map<String, CircuitBreakerConfig> resilienceCircuitBreakerConfig;
+  private final ResilienceCircuitBreakerProvider resilienceCircuitBreakerProvider;
+  private final Clock clock;
+
+  public ResilienceCircuitBreakerInterceptor(Config config, Clock clock) {
```

Comment:
The interceptor should take the parsed config rather than doing it internally. We want to leave it up to the circuit breaker user to hand us a config so they can, for example, replace it in tests.